espnet2/asr/frontend/s3prl.py

```diff
@@ -82,12 +87,26 @@ def _get_upstream(self, frontend_conf):
         ]:
             s3prl_upstream.model.encoder.layerdrop = 0.0
 
+            # check if adapter is added
+            if (
+                s3prl_upstream.model.__class__.__name__ == "Wav2Vec2Model"
+                and self.args.adapter_config
+            ):
+                s3prl_upstream = add_adapters_wav2vec2(
+                    s3prl_upstream, adapter_down_dim= self.args.adapter_config.adapter_down_dim, adapt_layers= eval(self.args.adapter_config.adapt_layers)
+                )
+
         from s3prl.upstream.interfaces import Featurizer
 
         if self.multilayer_feature:
             feature_selection = "hidden_states"
         else:
             feature_selection = "last_hidden_state"
+
+        # check if adapters are added and over-ride the feature selection
```

Comment:
@simpleoier to avoid dimension issues , when adapter is enabled ,` feature_selection ` is force set to "last_hidden_state"